src/Qir/qir-utils.ps1

```diff
@@ -93,10 +93,60 @@ function Build-CMakeProject {
     # Disable until the Catch header "src\Qir\Common\externals\catch2\catch.hpp" is updated to a version newer than v2.12.1 (from https://github.com/catchorg/Catch2).
     $warningFlags += " -Wno-extra-semi-stmt"    # https://clang.llvm.org/docs/DiagnosticsReference.html#wextra-semi-stmt
 
-
     $env:CFLAGS   += $warningFlags
     $env:CXXFLAGS += $warningFlags
 
+
+    # Sanitizers (https://clang.llvm.org/docs/UsersManual.html#controlling-code-generation):
+
+    $sanitizeFlags = "" 
+    if (-not ($IsWindows))
```

Comment:
Is this script intended for use with Windows PowerShell (5.1 and earlier, and as opposed to PowerShell for Windows, 6.0 and later)? If so, then the `$IsWindows` automatic variable was added in 6.0 such that a Windows user running this on 5.1 will fall through to this branch (`$null` coerces to `$false` under the `-not` operator). I'd suggest making sure that the `$IsWindows` variable exists unconditionally:

```powershell
if ($null -eq $IsWindows) {
    # PSEdition is "Desktop" for Windows PowerShell (5.1 and earlier),
    # and is "Core" for PowerShell Core (6.0) and for PowerShell
    # (7.0 and later).
    $IsWindows = $PSVersionTable.PSEdition -eq "Desktop";
}
```

Reply:
Our "[src/Qir/Runtime/README.md](https://github.com/microsoft/qsharp-runtime/blob/main/src/Qir/Runtime/README.md#prerequisites)" specifies as a prerequisite the "PowerShell Core or PowerShell 7+". That is why the `$IsWindows` automatic variable is always expected to be set (to `True` on Windows and to `False` otherwise).

The code inside of `if (-not ($IsWindows))` block is expected to be executed on non-Windows platforms (Linux and Mac).

I assume your question was based on assumption that my line `if (-not ($IsWindows))` was trying to work around different PowerShell versions, _all on Windows_. Thank you for the question. It reminded me how little I know about the PowerShell and encouraged me to read through the article [about_PowerShell_Editions](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_powershell_editions?view=powershell-7.1) and play a bit with the automatic variables on Win and WSL.

Reply:
If PowerShell 6.0 or later are already required, then please disregard my comment. We may separately want to make sure that there's a `#Requires` statement to communicate that to the user in an actionable way, but there's no issue with the code here as written if 6.0 or later are already required.